src/org/zaproxy/zap/extension/tokengen/ExtensionTokenGen.java

```diff
@@ -329,4 +334,37 @@ public URL getURL() {
 			return null;
 		}
 	}
+
+	private class SessionChangedListenerImpl implements SessionChangedListener {
+
+		@Override
+		public void sessionChanged(Session session) {
+		}
+
+		@Override
+		public void sessionAboutToChange(Session session) {
+			stopTokenGeneration();
+			generators = Collections.emptyList();
+
+			if (tokenPanel != null) {
+				tokenPanel.reset();
+			}
+
+			if (analyseTokensDialog != null) {
+				analyseTokensDialog.setVisible(false);
+			}
+
+			if (genTokensDialog != null) {
+				genTokensDialog.setVisible(false);
+			}
+		}
+
+		@Override
+		public void sessionScopeChanged(Session session) {
+		}
+
+		@Override
+		public void sessionModeChanged(Mode mode) {

```

Comment:
Should there be handling here? If the mode is switched safer and the target isnt in scope?